source/dub/dependency.d

```diff
@@ -555,16 +555,17 @@ unittest {
 
 	// Approximate versions.
 	a = Dependency("~>3.0");
-	b = Dependency(">=3.0.0 <4.0.0");
+	b = Dependency(">=3.0.0 <4.0.0-0");
 	assert(a == b, "Testing failed: " ~ a.toString());
 	assert(a.matches(Version("3.1.146")), "Failed: Match 3.1.146 with ~>0.1.2");
 	assert(!a.matches(Version("0.2.0")), "Failed: Match 0.2.0 with ~>0.1.2");
+	assert(!a.matches(Version("4.0.0-beta.1")));
```

Comment:
would be nice to have a check that checked it works for the minor version number, e.g. checking `"3.1.0-beta.1"` doesn't match `"~>3.0.0"`

Reply:
Added in line 569.